library_generation/generate_library.sh

```diff
@@ -182,6 +182,12 @@ case "${proto_path}" in
     removed_proto="google/rpc/http.proto"
     proto_files="${proto_files//${removed_proto}/}"
     ;;
+  "google/shopping"*)
+    # this proto is included in //google/shopping/css/v1:google-cloud-shopping-css-v1-java
+    # and //google/shopping/merchant/inventories/v1beta:google-cloud-merchant-inventories-v1beta-java
+    # and //google/shopping/merchant/reports/v1beta:google-cloud-merchant-reports-v1beta-java
+    proto_files="${proto_files} google/shopping/type/types.proto"
```

Comment:
Do we still need to include them here if these are included in java-common-protos?

Reply:
Some classes are missing in `proto-*` directory if `google/shopping/type/types.proto` is excluded.
```
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/Channel.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/ChannelOrBuilder.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/CustomAttribute.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/CustomAttributeOrBuilder.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/Destination.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/DestinationOrBuilder.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/Price.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/PriceOrBuilder.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/ReportingContext.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/ReportingContextOrBuilder.java
deleted:    java-shopping-css/proto-google-shopping-css-v1/src/main/java/com/google/shopping/type/TypesProto.java

```

So I think the answer to your question is no (these classes can be inherited from shared dependencies).